docs/DeveloperGuide.md

```diff
@@ -8,8 +8,189 @@
+the specified Command format. To deal with parsing parameters, spaces in University names and Module codes are to be replaced with underscores.
+
+The following class diagram illustrates the relationship between Command class and its subclasses as well as other classes related to Commands.
+![Command Class Diagram](./images/Command_class.png)
+
+#### Create Command
+
+A Create command can be used to create a university list and its corresponding timetable.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a create command.
+
+![Create Command Sequence Diagram](./images/CreateCommand_Sequence.png)
+
+#### Exit Command
+
+An Exit command can be used to exit the application.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of an exit command.
+
+![Exit Command Sequence Diagram](./images/ExitCommand_Sequence.png)
+
+#### Help Command
+
+A Help command can be used to exit the application.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a help command.
+
+![Help Command Sequence Diagram](./images/HelpCommand_Sequence.png)
+
+#### Add Command
+
+An add command can be used to add a lesson to the timetable or add a module mapping to the user university list.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of an add command.
+
+![Add Command Sequence Diagram](./images/AddCommand_Sequence.png)
+
+#### Delete Command
+
+A delete command can be used to delete a lesson from the timetable, delete a module mapping from the user university list or delete an entire user created university list.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a delete command.
+
+![Delete Command Sequence Diagram](./images/DeleteCommand_Sequence.png)
+
+#### View Command
+
+A view command can be used to view all user created university lists, view user's delete history, view user's selected university list or view all the user's created university lists' timetables.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a view command.
+
+![View Command Sequence Diagram](./images/ViewCommand_Sequence.png)
+
+#### List Command
+
+A list command can be used to display all the module mappings in the database, all the universities in the database, or allow users to filter by NUS module code or partner university name.
+
+The following class diagram illustrates the relationship between the respective classes involved in the creation and execution of a list command.
+
+![List Command Class Diagram](./images/ListCommand_Class.png)
```

Comment:
I like that there is an `enum` class shown together with the other classes to reinforce how the whole feature works all-together. Nice!